docker-compose.yml

```diff
@@ -2,6 +2,7 @@ version: '3.4'
 services:
   netbox: &netbox
     image: netboxcommunity/netbox:${VERSION-latest}
+    restart: unless-stopped
```

Comment:
Why did you go with `unless-stopped` instead of `on-failure`?

Reply:
Behaviour with `unless-stopped` is different compared to `on-failure`. 
With `on-stopped`:
1. killing a process or having a process exit != 0 will result in docker restarting the container.
2. Restarting docker daemon results in docker starting the containers
3. Restarting your server results in docker starting the containers
4. `docker-compose stop` results in docker not starting your containers after a docker daemon restart or server restart

`on-failure` does 1, and 4 in a similar fashion. But behaviour for 2 and 3 is different. 

Example of `on-failure` behaviour:
```
netbox-docker git/release *
root @ moby # docker-compose ps
            Name                           Command               State    Ports
---------------------------------------------------------------------------------
netbox-docker_netbox-worker_1   python3 /opt/netbox/netbox ...   Up
netbox-docker_netbox_1          /opt/netbox/docker-entrypo ...   Up
netbox-docker_nginx_1           nginx -c /etc/netbox-nginx ...   Up      80/tcp
netbox-docker_postgres_1        docker-entrypoint.sh postgres    Up      5432/tcp
netbox-docker_redis-cache_1     docker-entrypoint.sh sh -c ...   Up      6379/tcp
netbox-docker_redis_1           docker-entrypoint.sh sh -c ...   Up      6379/tcp

netbox-docker git/release *
root @ moby # docker container inspect netbox-docker_netbox_1 --format "{{.HostConfig.RestartPolicy.Name}}"
on-failure

netbox-docker git/release *
root @ moby # systemctl restart docker

netbox-docker git/release *
root @ moby # docker-compose ps
            Name                           Command               State    Ports
-------------------------------------------------------------------------------
netbox-docker_netbox-worker_1   python3 /opt/netbox/netbox ...   Exit 0
netbox-docker_netbox_1          /opt/netbox/docker-entrypo ...   Exit 0
netbox-docker_nginx_1           nginx -c /etc/netbox-nginx ...   Exit 0
netbox-docker_postgres_1        docker-entrypoint.sh postgres    Exit 0
netbox-docker_redis-cache_1     docker-entrypoint.sh sh -c ...   Exit 0
netbox-docker_redis_1           docker-entrypoint.sh sh -c ...   Exit 0
```


Example of `unless-stopped` behaviour:
```
netbox-docker git/release *
root @ moby # docker-compose up -d
Recreating netbox-docker_redis_1       ... done
Recreating netbox-docker_redis-cache_1   ... done
Recreating netbox-docker_postgres_1    ... done
Recreating netbox-docker_netbox-worker_1 ... done
Recreating netbox-docker_netbox_1        ... done
Recreating netbox-docker_nginx_1         ... done

netbox-docker git/release *
root @ moby # docker-compose ps
            Name                           Command               State    Ports
---------------------------------------------------------------------------------
netbox-docker_netbox-worker_1   python3 /opt/netbox/netbox ...   Up
netbox-docker_netbox_1          /opt/netbox/docker-entrypo ...   Up
netbox-docker_nginx_1           nginx -c /etc/netbox-nginx ...   Up      80/tcp
netbox-docker_postgres_1        docker-entrypoint.sh postgres    Up      5432/tcp
netbox-docker_redis-cache_1     docker-entrypoint.sh sh -c ...   Up      6379/tcp
netbox-docker_redis_1           docker-entrypoint.sh sh -c ...   Up      6379/tcp

netbox-docker git/release *
root @ moby # docker container inspect netbox-docker_netbox_1 --format "{{.HostConfig.RestartPolicy.Name}}"
unless-stopped

netbox-docker git/release *
root @ moby # systemctl restart docker

netbox-docker git/release *
root @ moby # docker-compose ps
            Name                           Command               State    Ports
---------------------------------------------------------------------------------
netbox-docker_netbox-worker_1   python3 /opt/netbox/netbox ...   Up
netbox-docker_netbox_1          /opt/netbox/docker-entrypo ...   Up
netbox-docker_nginx_1           nginx -c /etc/netbox-nginx ...   Up      80/tcp
netbox-docker_postgres_1        docker-entrypoint.sh postgres    Up      5432/tcp
netbox-docker_redis-cache_1     docker-entrypoint.sh sh -c ...   Up      6379/tcp
netbox-docker_redis_1           docker-entrypoint.sh sh -c ...   Up      6379/tcp
```


Reason for going `unless-stopped` is that you do not want your application to be dead after a docker daemon restart or server reboot.